lib/core-integration/src/Test/Integration/Scenario/API/Transactions.hs

```diff
@@ -1025,26 +1043,338 @@ spec = do
+            , expectListItemFieldEqual 0 status InLedger
+            , expectListItemFieldEqual 1 direction Incoming
+            , expectListItemFieldEqual 1 amount 1_000_000_000_000
+            , expectListItemFieldEqual 1 status InLedger
+            ]
+
+    -- This scenario covers the following matrix of cases. Cases where generated
+    -- using one of parwise test cases generation tools available online.
+    -- +----+----------+----------+------------+--------------+
+    --     |  start   |   end    |   order    |    result    |
+    -- +----+----------+----------+------------+--------------+
+    --   1 | edge     | edge     | ascending  | 2 ascending  |
+    --   2 | edge     | edge + 1 | descending | 2 descending |
+    --   3 | edge     | edge - 1 | empty      | 1st one      |
+    --   4 | edge     | empty    | empty      | 2 descending |
+    --   5 | edge + 1 | edge + 1 | empty      | 2nd one      |
+    --   6 | edge + 1 | edge - 1 | empty      | none         |
+    --   7 | edge + 1 | empty    | ascending  | 2nd one      |
+    --   8 | edge + 1 | edge     | descending | 2nd one      |
+    --   9 | edge - 1 | edge - 1 | ascending  | 1st one      |
+    --  10 | edge - 1 | empty    | descending | 2 descending |
+    --  11 | edge - 1 | edge     | empty      | 2 descending |
+    --  12 | edge - 1 | edge + 1 | empty      | 2 descending |
+    --  13 | empty    | empty    | empty      | 2 descending |
+    --  14 | empty    | edge     | empty      | 2 descending |
+    --  15 | empty    | edge + 1 | ascending  | 2 ascending  |
+    --  16 | empty    | edge - 1 | descending | 1st one      |
+    -- +----+----------+----------+------------+--------------+
+    it "TRANS_LIST_02,03 - Can limit/order results with start, end and order" $ \ctx -> do
+        (_, w, [(t1, a1), (t2, a2)]) <- fixtureWalletManyTxs ctx [10,20]
+
+        let toParam = toQueryParam . utcIso8601ToText
+        let matrix =
+                [
+                -- 1
+                  ( "?start="
+                    <> (toParam t1)
+                    <> "&end="
+                    <> (toParam t2)
+                    <> "&order=ascending"
+                  ,  [ expectListSizeEqual 2
+                     , expectListItemFieldEqual 0 amount a1
+                     , expectListItemFieldEqual 1 amount a2
+                     ]
+                  )
+                -- 2
+                , ( "?start="
+                    <> (toParam t1)
+                    <> "&end="
+                    <> (toParam $ addUTCTime 1 t2)
+                    <> "&order=descending"
+                  ,  [ expectListSizeEqual 2
+                     , expectListItemFieldEqual 0 amount a2
+                     , expectListItemFieldEqual 1 amount a1
+                     ]
+                  )
+                -- 3
+                , ( "?start="
+                    <> (toParam t1)
+                    <> "&end="
+                    <> (toParam $ addUTCTime (-1) t2)
```

Comment:
Similarly to the previous comment, assuming we want the closest time to `t2` that is smaller than `t2`, we have to use `pred 0` rather than `-1`:
```hs
<> (toParam $ addUTCTime (pred 0) t2)
```